lighthouse-core/lib/sd-validation/sd-validation.js

```diff
@@ -64,12 +62,73 @@ module.exports = async function validate(textInput) {
   if (schemaOrgErrors.length) {
     return schemaOrgErrors.map(error => {
       return {
-        validator: /** @type {ValidatorType} */ ('schema-org'),
+        validator: /** @type {LH.StructuredData.ValidatorType} */ ('schema-org'),
         path: error.path,
         message: error.message,
+        lineNumber: error.path ? getLineNumberFromJsonLDPath(inputObject, error.path) : null,
+        invalidTypes: error.invalidTypes,
       };
     });
   }
 
   return [];
 };
+
+/**
+ * @param {*} obj
+ * @param {string} path
+ * @returns null | number - line number of the path value in the prettified JSON
+ */
+function getLineNumberFromJsonLDPath(obj, path) {
+  // To avoid having an extra dependency on a JSON parser we set a unique key in the
+  // object and then use that to identify the correct line
+  const searchKey = Math.random().toString();
+  obj = JSON.parse(JSON.stringify(obj));
+
+  setValueAtJsonLDPath(obj, path, searchKey);
+  const jsonLines = JSON.stringify(obj, null, 2).split('\n');
```

Comment:
if we're reformatting the JSON then this line number is pretty much unrelated to the line number in the source right?

Reply:
Yeah. I tried to express that with `line number of the path value in the prettified JSON`, but do you have any suggestions to make it more explicit?

Reply:
Oh yeah yeah I think it's already clear in the code, I meant more from the users perspective!

Reply:
Yeah, that's a good point. Is there a strong need to address that though? The line numbers probably won't match anyway if the snippet is embedded in an HTML template.

Reply:
This is true, I'm just thinking there's not a ton of value in exposing the line number to the user then, but knowing where to highlight is massively awesome, so sorry for the distracting thread :)